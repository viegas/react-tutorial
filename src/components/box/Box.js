import React from 'react'
import styles from './Box.module.css'

function Box({children}) {
    return (
        <div className={styles.wrapper}>
            {children}
        </div>
    )
}

export default Box
