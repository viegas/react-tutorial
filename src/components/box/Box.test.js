import Box from './Box';

it('renders correctly', () => {
    const wrapper = shallow(<Box />);
    
    expect(wrapper.find('p')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly with children', () => {
    const wrapper = shallow(
        <Box>
            <p>Some children</p>
        </Box>
    );

    expect(wrapper.find('p')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
});
