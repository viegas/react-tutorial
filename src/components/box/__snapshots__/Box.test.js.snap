// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
/>
`;

exports[`renders correctly with children 1`] = `
<div
  className="wrapper"
>
  <p>
    Some children
  </p>
</div>
`;
